scripts/dev/k8sutil.py

```diff
@@ -0,0 +1,63 @@
+import time
+
+from kubernetes.client.rest import ApiException
+
+# time to sleep between retries
+SLEEP_TIME = 2
+# no timeout (loop forever)
+INFINITY = -1
+
+def _current_milliseconds():
+    return int(round(time.time() * 1000))
+
+def wait_for_condition(fn, condition, exceptions_to_ignore=[], codes_to_ignore=[],sleep_time=SLEEP_TIME, timeout=INFINITY) -> bool:
+    """
+    wait_for_condition accepts a function fn and a function condition,
+    it periodically calls the function fn and then applies the condition function on the result
+    until it returns True or we reach timeout
+
+    Exceptions raised by the call to fn are ignored if contained in exceptions_to_ignore
+    If ApiException is not ignored, if raised by the call to fn codes in codes_to_ignore are ignored
+    """
+    start_time = _current_milliseconds()
+    end = start_time + (timeout * 1000)
+
+
+    while _current_milliseconds() < end or timeout <= 0:
+        res = None
+        try:
+            res =  _ignore_error_codes(fn,codes_to_ignore)
+        except exceptions_to_ignore:
```

Comment:
[q] does excepting a list of exceptions work? Is it the same as `(ExceptionTypeA, ExceptionTypeB)` ?

Reply:
It actually works if passing a tuple, not a list. 

E.g. with this code:
```def f(i):
    if i==1:
        raise BufferError("Buffer")
    else:
        raise LookupError("Lookup")

def ignore_exc(ex = None):
    for i in range (0,2):
        try:
            f(i)
        except ex as e:
            print("EXCEPT: " + format(e))
```
the call to ` ignore_exc((BufferError,LookupError))` would work and print 

```
EXCEPT: Lookup
EXCEPT: Buffer
```

But the call to `ignore_exc([BufferError,LookupError])` would raise an exception:

```
TypeError: catching classes that do not inherit from BaseException is not allowed
```

I modified the comment to the function to make it clearer that a tuple should be used